### DISCUSSION TITLE
Add support for popper-content-wrapper styling

### DISCUSSION THREAD

**Discussion by eshelsil** [radix-ui/primitives#3892] [Ideas]
## Feature request
Mainly I would like to request the support for using Popper while using an "absolute" floating strategy rather than fixed.
Eventually, the popper-content-wrapper will get styled with "position: fixed;" and there is no straight-forward way to change it.

### Overview
In the codebase you can see that the strategy is hardcoded [`strategy: 'fixed',`](https://github.com/radix-ui/primitives/blob/main/packages/react/popper/src/Popper.tsx#L177)
which is the cause of the "position: fixed;" style.

So **all components that use the popper must use this "fixed" strategy** and it would be quite hard to choose the "absolute" strategy.

The main reason I had this need is due to the following issue:
Popover inside a scrollable container will appear outside of the container when exceeding its boundaries, even if its anchor is in the hidden overflown section of the container:

[Issue Example on Codesandbox](https://codesandbox.io/p/devbox/thirsty-cerf-qmnzh2?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clwbxxeuf00073b6lgntph5qc%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clwbxxeuf00023b6lu5qhrms2%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clwbxxeuf00043b6lu3zekd91%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clwbxxeuf00063b6l8zyhv163%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clwbxxeuf00023b6lu5qhrms2%2522%253A%257B%2522id%2522%253A%2522clwbxxeuf00023b6lu5qhrms2%2522%252C%2522activeTabId%2522%253A%2522clwbyz9g500023b6rqsp0xwbs%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwbxxeuf00013b6lracqf3nl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.html%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clwbycfhn00023b6lqfkb7ql4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Ftailwind.config.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.codesandbox%252Ftasks.json%2522%252C%2522id%2522%253A%2522clwbyo0qf007b3b6r9zs39h77%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clwbyr8pf00023b6rgjeim8rq%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A1%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A1%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FMyPopover.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clwbyz9g500023b6rqsp0xwbs%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A24%252C%2522startColumn%2522%253A22%252C%2522endLineNumber%2522%253A24%252C%2522endColumn%2522%253A22%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clwbxxeuf00063b6l8zyhv163%2522%253A%257B%2522id%2522%253A%2522clwbxxeuf00063b6l8zyhv163%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clwbz4nhs006z3b6rlisexbma%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clwbz4nhs006z3b6rlisexbma%2522%257D%252C%2522clwbxxeuf00043b6lu3zekd91%2522%253A%257B%2522id%2522%253A%2522clwbxxeuf00043b6lu3zekd91%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwbz3r6k00273b6rgzl1kms7%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clwbz3reb001adji20ysx22m3%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clwbz3r6k00273b6rgzl1kms7%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)
<img width="490" alt="Screenshot 2024-05-18 at 14 27 34" src="https://github.com/radix-ui/primitives/assets/10271715/3de21708-2fdb-4a2b-b90c-34e81cc3d1ca">

### Who does this impact? Who is this for?

I think it might impact everyone

### Additional context
In order to solve that issue I had to add {absolute: !important} to css because the data-radix-popper-content-wrapper's fixed position is set as a style:
```
.absolute-popover-container > [data-radix-popper-content-wrapper] {
  position: absolute !important;
  pointer-events: all;
}

```

[Hack Solution Example in Codesandbox](https://codesandbox.io/p/devbox/sweet-rui-q8t4kv?file=%2Fsrc%2FApp.tsx%3A48%2C50&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clwbxxeuf00073b6lgntph5qc%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clwbxxeuf00023b6lu5qhrms2%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clwbxxeuf00043b6lu3zekd91%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clwbxxeuf00063b6l8zyhv163%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B90%252C10%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clwbxxeuf00023b6lu5qhrms2%2522%253A%257B%2522id%2522%253A%2522clwbxxeuf00023b6lu5qhrms2%2522%252C%2522activeTabId%2522%253A%2522clwbyz9g500023b6rqsp0xwbs%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwbxxeuf00013b6lracqf3nl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.html%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clwbycfhn00023b6lqfkb7ql4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Ftailwind.config.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.codesandbox%252Ftasks.json%2522%252C%2522id%2522%253A%2522clwbyo0qf007b3b6r9zs39h77%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clwbyr8pf00023b6rgjeim8rq%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A1%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A1%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FMyPopover.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clwbyz9g500023b6rqsp0xwbs%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A48%252C%2522startColumn%2522%253A50%252C%2522endLineNumber%2522%253A48%252C%2522endColumn%2522%253A50%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clwbxxeuf00063b6l8zyhv163%2522%253A%257B%2522id%2522%253A%2522clwbxxeuf00063b6l8zyhv163%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clwbz4nhs006z3b6rlisexbma%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clwbz4nhs006z3b6rlisexbma%2522%257D%252C%2522clwbxxeuf00043b6lu3zekd91%2522%253A%257B%2522id%2522%253A%2522clwbxxeuf00043b6lu3zekd91%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwbz3r6k00273b6rgzl1kms7%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clwbz3reb001adji20ysx22m3%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clwbz3r6k00273b6rgzl1kms7%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)
<img width="415" alt="Screenshot 2024-05-18 at 14 24 12" src="https://github.com/radix-ui/primitives/assets/10271715/20847dfa-9f76-46e4-889d-b3a5ec75186c">

**Comment by mrdavidjcole** [comment]
Just ran into this issue today as well and found myself here after some googling. Would love to be able to either style it directly, or pass a prop to make it `position: absolute` instead of `position: fixed`.

**Comment by JUDE-EQ** [comment]
+1 on this, give us some control on `data-radix-popper-content-wrapper`

**Comment by christian-reichart** [comment]
+1 on this, I want to be able to style the options like a popover, so it behaves kinda like a native mobile dropdown

**Comment by christian-reichart** [comment]
if anybody cares I wrote this patch (with patch-package) to allow me to set a className to the wrapper (because I use Tailwind, same could be done with style). I also added a position mode called "modal" that takes the whole screen and centers the options.

This could be done cleaner for sure, I just need a fix quickly for now. :)

```diff
diff --git a/node_modules/@radix-ui/react-select/dist/index.d.ts b/node_modules/@radix-ui/react-select/dist/index.d.ts
index 9775d79..11c7cdf 100644
--- a/node_modules/@radix-ui/react-select/dist/index.d.ts
+++ b/node_modules/@radix-ui/react-select/dist/index.d.ts
@@ -78,7 +78,8 @@ interface SelectContentImplProps extends Omit<SelectPopperPositionProps, keyof S
      * Can be prevented.
      */
     onPointerDownOutside?: DismissableLayerProps['onPointerDownOutside'];
-    position?: 'item-aligned' | 'popper';
+    wrapperClassName?: string;
+    position?: 'item-aligned' | 'popper' | 'modal';
 }
 interface SelectItemAlignedPositionProps extends PrimitiveDivProps, SelectPopperPrivateProps {
 }
diff --git a/node_modules/@radix-ui/react-select/dist/index.js b/node_modules/@radix-ui/react-select/dist/index.js
index 25d9296..603225c 100644
--- a/node_modules/@radix-ui/react-select/dist/index.js
+++ b/node_modules/@radix-ui/react-select/dist/index.js
@@ -461,7 +461,7 @@ var SelectContentImpl = React.forwardRef(
       },
       [context.value]
     );
-    const SelectPosition = position === "popper" ? SelectPopperPosition : SelectItemAlignedPosition;
+    const SelectPosition = position === "popper" ? SelectPopperPosition : position === "modal" ? SelectModalPosition : SelectItemAlignedPosition;
     const popperContentProps = SelectPosition === SelectPopperPosition ? {
       side,
       sideOffset,
@@ -563,7 +563,7 @@ var SelectContentImpl = React.forwardRef(
 SelectContentImpl.displayName = CONTENT_IMPL_NAME;
 var ITEM_ALIGNED_POSITION_NAME = "SelectItemAlignedPosition";
 var SelectItemAlignedPosition = React.forwardRef((props, forwardedRef) => {
-  const { __scopeSelect, onPlaced, ...popperProps } = props;
+  const { __scopeSelect, onPlaced, wrapperClassName, ...popperProps } = props;
   const context = useSelectContext(CONTENT_NAME, __scopeSelect);
   const contentContext = useSelectContentContext(CONTENT_NAME, __scopeSelect);
   const [contentWrapper, setContentWrapper] = React.useState(null);
@@ -693,6 +693,7 @@ var SelectItemAlignedPosition = React.forwardRef((props, forwardedRef) => {
             position: "fixed",
             zIndex: contentZIndex
           },
+          className: wrapperClassName,
           children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
             import_react_primitive.Primitive.div,
             {
@@ -714,12 +715,58 @@ var SelectItemAlignedPosition = React.forwardRef((props, forwardedRef) => {
   );
 });
 SelectItemAlignedPosition.displayName = ITEM_ALIGNED_POSITION_NAME;
+var MODAL_POSITION_NAME = "SelectModalPosition";
+var SelectModalPosition = React.forwardRef((props, forwardedRef) => {
+  const { __scopeSelect, onPlaced, wrapperClassName, ...popperProps } = props;
+  const [contentWrapper, setContentWrapper] = React.useState(null);
+  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
+    SelectViewportProvider,
+    {
+      scope: __scopeSelect,
+      contentWrapper,
+      shouldExpandOnScrollRef,
+      onScrollButtonChange: handleScrollButtonChange,
+      children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
+        "div",
+        {
+          ref: setContentWrapper,
+          style: {
+            display: "flex",
+            flexDirection: "column",
+            justifyContent: "center",
+            alignItems: "center",
+            position: "fixed",
+            left: 0,
+            top: 0,
+            width: "100vw",
+            height: "100vh",
+            zIndex: 100,
+          },
+          className: wrapperClassName,
+          children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
+            import_react_primitive.Primitive.div,
+            {
+              ...popperProps,
+              ref: composedRefs,
+              style: popperProps.style,
+            }
+          )
+        }
+      )
+    }
+  );
+});
+SelectModalPosition.displayName = MODAL_POSITION_NAME;
+
+
 var POPPER_POSITION_NAME = "SelectPopperPosition";
 var SelectPopperPosition = React.forwardRef((props, forwardedRef) => {
   const {
     __scopeSelect,
     align = "start",
     collisionPadding = CONTENT_MARGIN,
+    wrapperClassName,
+    className,
     ...popperProps
   } = props;
   const popperScope = usePopperScope(__scopeSelect);
@@ -730,6 +777,7 @@ var SelectPopperPosition = React.forwardRef((props, forwardedRef) => {
       ...popperProps,
       ref: forwardedRef,
       align,
+      className: [className, wrapperClassName].filter(Boolean).join(" "),
       collisionPadding,
       style: {
         // Ensure border-box for floating-ui calculations
diff --git a/node_modules/@radix-ui/react-select/dist/index.mjs b/node_modules/@radix-ui/react-select/dist/index.mjs
index 5d874b7..d8cc59c 100644
--- a/node_modules/@radix-ui/react-select/dist/index.mjs
+++ b/node_modules/@radix-ui/react-select/dist/index.mjs
@@ -394,7 +394,7 @@ var SelectContentImpl = React.forwardRef(
       },
       [context.value]
     );
-    const SelectPosition = position === "popper" ? SelectPopperPosition : SelectItemAlignedPosition;
+    const SelectPosition = position === "popper" ? SelectPopperPosition : position === "modal" ? SelectModalPosition : SelectItemAlignedPosition;
     const popperContentProps = SelectPosition === SelectPopperPosition ? {
       side,
       sideOffset,
@@ -496,7 +496,7 @@ var SelectContentImpl = React.forwardRef(
 SelectContentImpl.displayName = CONTENT_IMPL_NAME;
 var ITEM_ALIGNED_POSITION_NAME = "SelectItemAlignedPosition";
 var SelectItemAlignedPosition = React.forwardRef((props, forwardedRef) => {
-  const { __scopeSelect, onPlaced, ...popperProps } = props;
+  const { __scopeSelect, onPlaced, wrapperClassName, ...popperProps } = props;
   const context = useSelectContext(CONTENT_NAME, __scopeSelect);
   const contentContext = useSelectContentContext(CONTENT_NAME, __scopeSelect);
   const [contentWrapper, setContentWrapper] = React.useState(null);
@@ -624,7 +624,8 @@ var SelectItemAlignedPosition = React.forwardRef((props, forwardedRef) => {
             display: "flex",
             flexDirection: "column",
             position: "fixed",
-            zIndex: contentZIndex
+            zIndex: contentZIndex,
+            className: wrapperClassName,
           },
           children: /* @__PURE__ */ jsx(
             Primitive.div,
@@ -647,12 +648,58 @@ var SelectItemAlignedPosition = React.forwardRef((props, forwardedRef) => {
   );
 });
 SelectItemAlignedPosition.displayName = ITEM_ALIGNED_POSITION_NAME;
+var MODAL_POSITION_NAME = "SelectModalPosition";
+var SelectModalPosition = React.forwardRef((props, forwardedRef) => {
+  const { __scopeSelect, onPlaced, wrapperClassName, ...popperProps } = props;
+  const [content, setContent] = React.useState(null);
+  const [contentWrapper, setContentWrapper] = React.useState(null);
+  const composedRefs = useComposedRefs(forwardedRef, (node) => setContent(node));
+  return /* @__PURE__ */ jsx(
+    SelectViewportProvider,
+    {
+      scope: __scopeSelect,
+      contentWrapper,
+      shouldExpandOnScrollRef: false,
+      onScrollButtonChange: () => {},
+      children: /* @__PURE__ */ jsx(
+        "div",
+        {
+          ref: setContentWrapper,
+          style: {
+            display: "flex",
+            flexDirection: "column",
+            justifyContent: "center",
+            alignItems: "center",
+            position: "fixed",
+            left: 0,
+            top: 0,
+            width: "100vw",
+            height: "100vh",
+            zIndex: 100,
+          },
+          className: wrapperClassName,
+          children: /* @__PURE__ */ jsx(
+            Primitive.div,
+            {
+              ...popperProps,
+              ref: composedRefs,
+              style: popperProps.style,
+            }
+          )
+        }
+      )
+    }
+  );
+});
+SelectModalPosition.displayName = MODAL_POSITION_NAME;
 var POPPER_POSITION_NAME = "SelectPopperPosition";
 var SelectPopperPosition = React.forwardRef((props, forwardedRef) => {
   const {
     __scopeSelect,
     align = "start",
     collisionPadding = CONTENT_MARGIN,
+    wrapperClassName,
+    className,
     ...popperProps
   } = props;
   const popperScope = usePopperScope(__scopeSelect);
@@ -661,6 +708,7 @@ var SelectPopperPosition = React.forwardRef((props, forwardedRef) => {
     {
       ...popperScope,
       ...popperProps,
+      className: [className, wrapperClassName].filter(Boolean).join(" "),
       ref: forwardedRef,
       align,
       collisionPadding,

```

**Comment by RomanSkrypnik** [comment]
Facing the same issue

**Comment by ivkrpv** [comment]
a possible workaround (here it changes the wrapper's styles on small screens):
```
@media not all and (min-width: 640px) {
  [data-radix-popper-content-wrapper] {
    position: absolute !important;
    bottom: 1rem !important;
    left: 1rem !important;
    top: 1rem !important;
    transform: none !important;
  }
}
```

PS: but this will change all your popovers
PS2: ended up adding this rule dynamically on popover opening

```
function fixMobilePopover(open: boolean) {
  const ID = 'popover-fix';

  if (open) {
    const style = document.createElement('style');

    style.id = ID;
    style.innerHTML = `
      @media (max-width: 640px) {
        [data-radix-popper-content-wrapper] {
          position: absolute !important;
          bottom: 1rem !important;
          left: unset !important;
          right: 1rem !important;
          top: 1rem !important;
          transform: none !important;
        }
      }
    `;

    document.head.appendChild(style);
  } else {
    const style = document.getElementById(ID);

    if (style) {
      // wait for the popover to close before removing the style
      setTimeout(() => {
        document.head.removeChild(style);
      }, 150);
    }
  }
}
```
in your react component:
```
useEffect(() => {
    fixMobilePopover(isOpen);
  }, [isOpen]);
```

**Comment by samuel99-c** [comment]
A workaround using tailwind:

tailwind.config.js:
```
plugins: [
      function ({ addVariant }) {
        addVariant(
          'radix-popper-wrapper',
          '& [data-radix-popper-content-wrapper]'
        );
      },
    ],
```


The component:

```
<Popover.Root>
        <div
          ref={containerRef}
          className="radix-popper-wrapper:!absolute radix-popper-wrapper:!transform-none radix-popper-wrapper:!right-0 radix-popper-wrapper:!top-12 relative"
        >
          <Popover.Trigger asChild>
            <button>
             TRIGGER
            </button>
          </Popover.Trigger>
          <Popover.Portal container={containerRef.current}>
            <Popover.Content
            >
              CONTENT
            </Popover.Content>
          </Popover.Portal>
        </div>
      </Popover.Root>
```

**Comment by chrislgrigg** [comment]
Is there a reason it's not exposed? It seems like a very obvious change... Is there some implementation detail that's hidden to keep folks out of trouble?

**Comment by kachurun** [comment]
I need to add some classes to this wrapper, what seems not possible too, thats weird

**Comment by ashwinchandran13** [comment]
Alright I got a solution for this (workaround). I'm still testing the coverage of effect of this solution but to minimize the effect of it I had to introduce a class: 
```js 
// css
 .DropdownTbody 
[data-radix-popper-content-wrapper] { 
   position: absolute !important;
} 
// jsx 
<tbody className='relative DropdownTbody'>  {/* placing class here /*} 
  <DropdownMenu open={filterMenuOpen} onOpenChange={setFilterMenuOpen}> 
      <DropdownMenuContent 
           className="DropdownMenuContent outline-none drop-shadow-2xl shadow-2xl bg-[#1A1A1A] z-50"
           alignOffset={9} 
           sideOffset={10} 
           align="end" 
           side='bottom' 
  > 
        <DropdownMenuItem className="outline-none"> 
            <div className="flex gap-2 text-xs items-center p-4 bg-[#1A1A1A] text-white hover:cursor-pointer hover:bg-[#333] absolute z-50 min-w-fit w-[250px]" > 
                Filter menu 
            </div> {/* making this div absolute to the tbody of a table which is now relative /*}
        </DropdownMenuItem> 
     </DropdownMenuContent> 
  </DropdownMenu> 
</tbody> 
```

**Comment by lucasmcht** [comment]
If you are using **tailwind**, I have found this workaround for now, by wrapping the element using the `PopperContent` with this div : 
```jsx
//Replace 'my-0' by the style you want to apply
<div className='[&_[data-radix-popper-content-wrapper]]:my-0'>
 {/*... your popper element*/}
</div>
```


Example with Popover (i'm using shadcn):
```jsx
  <div className='[&_[data-radix-popper-content-wrapper]]:my-0'>
    <Popover>
      <PopoverTrigger asChild>
        <Button
          variant='outline'
          role='combobox'
          aria-expanded={false}
          className='w-full justify-between'
        >
          Open Popover
        </Button>
      </PopoverTrigger>
      <PopoverContent className='mt-0 w-[var(--radix-popover-trigger-width)] p-0'>
        Blah blah blah
      </PopoverContent>
    </Popover>
  </div>
```

**Comment by GabrielFeliciano** [comment]
> If you are using **tailwind**, I have found this workaround for now, by wrapping the element using the `PopperContent` with this div :
> 
> //Replace 'my-0' by the style you want to apply
> <div className='[&_[data-radix-popper-content-wrapper]]:my-0'>
>  {/*... your popper element*/}
> </div>
> Example with Popover (i'm using shadcn):
> 
>   <div className='[&_[data-radix-popper-content-wrapper]]:my-0'>
>     <Popover>
>       <PopoverTrigger asChild>
>         <Button
>           variant='outline'
>           role='combobox'
>           aria-expanded={false}
>           className='w-full justify-between'
>         >
>           Open Popover
>         </Button>
>       </PopoverTrigger>
>       <PopoverContent className='mt-0 w-[var(--radix-popover-trigger-width)] p-0'>
>         Blah blah blah
>       </PopoverContent>
>     </Popover>
>   </div>

I've found a alternative that don't require a top level `<div />`

```jsx
<div className="[[data-radix-popper-content-wrapper]:has(&)]:min-w-56">
 {/*... your popper element*/}
</div>
```

Replace `min-w-56` with any utility

**Comment by Naufal05R** [comment]
Agree with this

**Comment by rshelnutt** [comment]
Somewhat related, but apparently a different flavor required for shadcn:

```jsx
<DropdownMenuContent align="end" className="w-[var(--radix-popper-anchor-width)]">
   {/* ... */}
</DropdownMenuContent>
```

**Comment by janwirth** [comment]
Still an issue.

**Comment by Dari4sho** [comment]
I found this to be a problem especially with dropdowns. `data-radix-popper-content-wrapper` comes with z-index: 50, meaning you are basically limited to 50 if you generally want dropdowns to be on top - otherwise you need to resort to one of the given hacks. So if you have another popper-based component (e.g. HoverCard with shadcn), which doesn't always need to be the top level element, you quickly run into conflicts.

**Comment by anasSPZ** [comment]
Still an issue, and would be good to be able to change `data-radix-popper-content-wrapper` class directly.

**Comment by Alletkla** [comment]
I also find this to be an issue on Popovers, they move around slightly on scroll and can completely overlap the trigger on elastic scroll / rubber banding (quickly scrolling on trackpad or touch devices)

It does not happen on position absolute, so I wonder why fixed is used. 

https://www.radix-ui.com/primitives/docs/components/popover
<img width="1352" height="741" alt="Image" src="https://github.com/user-attachments/assets/5bb98b48-db05-449c-a42e-74cba5bc9839" />

**Comment by iamrane** [comment]
I added this on a parent element to add styling on it. (with tailwind)

`[&_[data-radix-popper-content-wrapper]]:your-class`
